rust-runtime/aws-smithy-types/src/instant/format.rs

```diff
@@ -490,82 +393,193 @@ pub(crate) mod rfc3339 {
 }
 
 #[cfg(test)]
-mod test {
-    use super::rfc3339::format;
+mod tests {
+    use super::*;
     use crate::Instant;
-    use proptest::proptest;
+    use lazy_static::lazy_static;
+    use proptest::prelude::*;
+
+    #[derive(Debug, serde::Deserialize)]
+    struct TestCase {
+        epoch_seconds: i64,
+        epoch_subsecond_nanos: u32,
+        iso: String,
+        formatted: Option<String>,
+    }
+    impl TestCase {
+        fn time(&self) -> Instant {
+            Instant::from_secs_and_nanos(self.epoch_seconds, self.epoch_subsecond_nanos)
+        }
+    }
+
+    #[derive(serde::Deserialize)]
+    struct TestCases {
+        format_date_time: Vec<TestCase>,
+        format_http_date: Vec<TestCase>,
+        format_epoch_seconds: Vec<TestCase>,
+        parse_date_time: Vec<TestCase>,
+        parse_http_date: Vec<TestCase>,
+        parse_epoch_seconds: Vec<TestCase>,
+    }
+
+    lazy_static! {
+        static ref TEST_CASES: TestCases = {
+            // This test suite can be regenerated by the following Kotlin class:
+            // `codegen/src/test/kotlin/software/amazon/smithy/rust/tool/TimeTestSuiteGenerator.kt`
+            let json = include_str!("../../test_data/date_time_format_test_suite.json");
```

Comment:
_slight_ preference for reading it from the filesystem because it can be changed without recompiling